File: src/main/java/org/tud/oas/api/fca/FCAGeoJSONResponse.java
package org.tud.oas.api.fca;

import org.locationtech.jts.geom.Coordinate;
import org.tud.oas.population.Population;

public class FCAGeoJSONResponse {
    public String type = "FeatureCollection"; 
    public GeoJsonFeature[] features;
    
    public FCAGeoJSONResponse(Population population, float[] weights)
    {
        GeoJsonFeature[] points = new GeoJsonPoint[weights.length];
        for (int i=0; i<points.length; i++) 
        {
            Coordinate p = population.getPoint(i);
            points[i] = new GeoJsonPoint((int)weights[i], p);
        }
        this.features = points;
    }

}

File: src/api/fca/FCAGeoJSONResponse.cs
using System;
using NetTopologySuite.Geometries;
using DVAN.Population;

namespace DVAN.API
{
    public class FCAGeoJSONResponse
    {
        public String type { get; set; }
        public GeoJsonFeature[] features { get; set; }

        public FCAGeoJSONResponse(PopulationContainer population, float[] weights)
        {
            this.type = "FeatureCollection";
            GeoJsonFeature[] points = new GeoJsonPoint[weights.Length];
            for (int i = 0; i < points.Length; i++) {
                Coordinate p = population.getPoint(i);
                points[i] = new GeoJsonPoint((int)weights[i], p);
            }
            this.features = points;
        }

    }
}
